src/sentry/models/grouphashmetadata.py

```diff
@@ -0,0 +1,64 @@
+from typing import Any
+
+from django.db import models
+from django.utils import timezone
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import FlexibleForeignKey, JSONField, Model, region_silo_model
+
+
+@region_silo_model
+class GroupHashMetadata(Model):
+    __relocation_scope__ = RelocationScope.Excluded
+
+    class GroupingMethod(models.TextChoices):
+        HASH = "hash"
+        SEER = "seer"
+
+    class HashBasis(models.TextChoices):
+        MESSAGE = "msg"
+        FULL_STACKTRACE = "stack"
+        APP_STACKTRACE = "app_stack"
+        FINGERPRINT = "fingerprint"
+        # TODO: Are there others?
```

Comment:
Custom fingerprint.

Today I've found [this](https://sentry.sentry.io/issues/4956614130/?project=1&query=issue.category%3A%5Bperformance%2Ccron%2Creplay%5D&referrer=issue-stream&statsPeriod=7d&stream_index=0/) new way of grouping
<img width="395" alt="image" src="https://github.com/user-attachments/assets/1d16f876-c88c-4c26-969c-8a74d8e677a6">

Reply:
Yeah, I actually think I need to go through all of the strategies, variants, and components to figure out which of them can turn into an over-arching reason for grouping. Will definitely take some more research, but I'll cross that bridge when I come to it.